solr/core/src/java/org/apache/solr/handler/component/SearchHandler.java

```diff
@@ -126,19 +125,24 @@ public class SearchHandler extends RequestHandlerBase
   private PluginInfo shfInfo;
   private SolrCore core;
 
+  /**
+   * The default set of components that every handler gets. You can change this by defining the
+   * specific components for a handler. It puts the {@link QueryComponent} first as subsequent
+   * components assume that the QueryComponent ran and populated the document list.
+   *
+   * @return A list of component names.
+   */
   protected List<String> getDefaultComponents() {
-    ArrayList<String> names = new ArrayList<>(9);
-    names.add(QueryComponent.COMPONENT_NAME);
-    names.add(FacetComponent.COMPONENT_NAME);
-    names.add(FacetModule.COMPONENT_NAME);
-    names.add(MoreLikeThisComponent.COMPONENT_NAME);
-    names.add(HighlightComponent.COMPONENT_NAME);
-    names.add(StatsComponent.COMPONENT_NAME);
-    names.add(DebugComponent.COMPONENT_NAME);
-    names.add(ExpandComponent.COMPONENT_NAME);
-    names.add(TermsComponent.COMPONENT_NAME);
-
-    return names;
+    List<String> l = new ArrayList<String>(SearchComponent.STANDARD_COMPONENTS.keySet());
```

Comment:
As a result RTG occurs as a default component that causes a problem in an essential cloud test. 😭

Reply:
humm...    Maybe I just back out this optimization...

Reply:
check out the changes I made to back this out, but still keep the name `STANDARD_COMPONENTS`...